### DISCUSSION TITLE
Making the first contribution

### DISCUSSION THREAD

**Discussion by jpangas** [mozilla/bugbug#3342] [Q&A] [answered]
How can I make my first contribution to the `bugbug` repository?

**Comment by jpangas** [comment, marked as answer]
## Making your contribution

Ensure you have properly setup the `bugbug` repository on your computer. Please see #3338 if you would like to know how to properly setup the repository on your computer. 

1. Synchronize any changes that could have occurred before you start working on any contributions.
<pre>
git checkout main
git fetch upstream
git merge upstream/main
</pre>

2. Create a branch which you shall use to make your changes/add your contribution. First time contributors can start with [Issues marked as `good-first-bug`](https://github.com/mozilla/bugbug/labels/good-first-bug). Issues are considered not assigned, until there is a PR linked to them. Feel free to work on any unassigned issue, you don't need to ask first. Remember to always use a feature_branch to make your changes. It's good practice to never work on the main/master branch.
```bash
git checkout -b my_feature_branch_name
```
3. Confirm you have `pre-commit` installed on your working environment by running:
```bash
pre-commit -V #Output should be pre-commit version
```

- `pre-commit`  runs checks on your code before a commit thus improves the quality of commits. It should have been installed when you ran the command `pip install -r test-requirements.txt` from the [**Setup and Prerequisities**](https://github.com/mozilla/bugbug#setup-and-prerequisites) in `README.md`

4. Make your changes on your feature branch and when you are done editing, add the changed files and commit them.
<pre>
git add modified_file
git commit -m "Type a nice commit message"
</pre>
- `pre-commit` will run code style checks whenever you commit your changes and if all checks have passed, the commit will complete.

5.  Push the changes to your forked repository (origin) on your GitHub Account:
```bash
git push -u origin my_feature_branch_name
```
6. Make a pull request on Github. You can follow this [guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) on how to make a pull request from a forked repository. Please also see #3340 on how to submit your pull request to `bugbug`

**Reply by AngieMeliss** [reply]
Hi @jpangas , how are you?
when I ran the `pre-commit ` the output was `pre-commit 3.4.0`, is this normal?

**Reply by jpangas** [reply]
Currently `bugbug` uses version `3.5.0`. Have you pulled the latest changes from the repository?

**Reply by AngieMeliss** [reply]
Yes, I updated my master branch with `git fetch upstream` and `git merge upstream/master`
I went to the branch where I want to make the changes and to verify I used `git diff`, and there is no difference.
`pre-commit -V` outputs `pre-commit 3.4.0`

**Reply by suhaibmujahid** [reply]
@AngieMeliss the version for `pre-commit` is not a big deal. Anyway, you could update it if you want by installing the test dependencies:
```
pip install -r test-requirements.txt
```
